app/presenters/idv/by_mail/letter_enqueued_presenter.rb

```diff
@@ -0,0 +1,68 @@
+module Idv
+  module ByMail
+    class LetterEnqueuedPresenter
+      include ActionView::Helpers::TranslationHelper
+      include Rails.application.routes.url_helpers
+
+      attr_reader :url_options
+
+      def initialize(idv_session:, current_user:, user_session:, url_options:)
+        @idv_session = idv_session
+        @current_user = current_user
+        @user_session = user_session
+        @url_options = url_options
+      end
+
+      def address_lines
+        [
+          pii[:address1],
+          pii[:address2],
+          "#{pii[:city]}, #{pii[:state]} #{pii[:zipcode]}",
+        ].compact
+      end
+
+      def button_text
+        if sp
+          t('idv.cancel.actions.exit', app_name: APP_NAME)
+        else
+          t('idv.buttons.continue_plain')
+        end
+      end
+
+      def button_destination
+        if sp
+          return_to_sp_cancel_path(step: :get_a_letter, location: :come_back_later)
+        else
+          account_path
+        end
+      end
+
+      private
+
+      attr_accessor :idv_session, :user_session
+
+      def sp
+        @sp ||= idv_session.service_provider
+      end
+
+      def pii
```

Comment:
I think we should add some test coverage for this to the presenter spec, just verifying that PII can be loaded from any of these places and the order that it is preferred

Reply:
Agreed. Added.